force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -288,8 +386,15 @@ public with sharing class FlowControlLoopsCollections {
     public static Decimal countMoney() {
         // The list of money in the wallet, represented in dollars and cents
         List<Decimal> moneyInWallet = new List<Decimal>{0.50, 10, 3.84, 24.60, 9.08, 50, 4.90}; //DO NOT CHANGE
-
-        return null; // Replace null with the variable you used to store the result
+        decimal result = 0;
+        for ( Decimal i : moneyInWallet) {
+            if ( result < 40){
+                result += i;
+                 } else {
+                    break;
+                    }    
+        }
+        return result; // Replace null with the variable you used to store the result
```

Comment:
Good work. I am going to share my solution below. I am a fan of writing the least amount of conditional logic we can. It is very hard to reason about. It is almost the same, but I think it is a little more clear. 

```
  public static Decimal countMoney() {
        // The list of money in the wallet, represented in dollars and cents
        List<Decimal> moneyInWallet = new List<Decimal>{0.50, 10, 3.84, 24.60, 9.08, 50, 4.90}; //DO NOT CHANGE
        Decimal total = 0;
        for(Decimal money : moneyInWallet){
            total += money;
            if(total > 40){
                break;
            }
        }
        return total; // Replace null with the variable you used to store the result
    }
 ```